product-service/src/main/java/com/extremecoder/productservice/controller/ProductController.java

```diff
@@ -2,24 +2,23 @@
 
 import com.extremecoder.productservice.model.Product;
 import com.extremecoder.productservice.service.ProductService;
-import org.springframework.beans.factory.annotation.Autowired;
+import com.extremecoder.productservice.utils.UrlConstraint;
+import lombok.RequiredArgsConstructor;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
 import java.util.List;
 
 @RestController
-@RequestMapping("/product")
+@RequestMapping(UrlConstraint.ProductManagement.ROOT)
+@RequiredArgsConstructor
 public class ProductController {
 
-    @Autowired
-    private ProductService productService;
+    private final ProductService productService;
 
-    @GetMapping("/")
+    @GetMapping
     public List<Product> getList() {
         return productService.getList();
```

Comment:
Can not we improve the method name as `getProducts()`? It matches or reflects the context more i think.

Reply:
Yes bhai, you are right. But as last discussion we can't touch anything which is not my task at ticket. Hope you understand my point

Reply:
Yes. i understood. Thanks bhai.